lib/core/client.js

```diff
@@ -972,6 +973,15 @@ function _resume (client, sync) {
         util.destroy(socket)
         return
       }
+
+      if (
+        client[kHeadersTimeout] &&
+        socket &&
+        !socket[kWriting] &&
+        !socket[kParser].timeout
+      ) {
+        socket[kParser].timeout = setTimeout(onHeadersTimeout, client[kHeadersTimeout], socket[kParser])
+      }
```

Comment:
I don't fully understand the move of this check right here (It is more a question than a review comment). I understand that by checking if the `socket` is writing (the body), we can skip the timeout setup.

Reply:
We have to move it before `pending === 0` check.

Reply:
Got it, thanks.